src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -15,118 +16,198 @@
     }
-    return CHIP_NO_ERROR;
+    else
+    {
+        value.SetNonNull(Traits::StorageToWorking(temp));
+    }
+    return status;
 }
 
-} // anonymous namespace
-
-bool emberAfModeSelectClusterChangeToModeCallback(CommandHandler * commandHandler, const ConcreteCommandPath & commandPath,
-                                                  const ModeSelect::Commands::ChangeToMode::DecodableType & commandData)
+EmberAfStatus Instance::SetStartUpMode(uint8_t value) const
 {
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "ModeSelect: Entering emberAfModeSelectClusterChangeToModeCallback");
-    EndpointId endpointId = commandPath.mEndpointId;
-    uint8_t newMode       = commandData.newMode;
-    // Check that the newMode matches one of the supported options
-    const ModeSelect::Structs::ModeOptionStruct::Type * modeOptionPtr;
-    Status checkSupportedModeStatus =
-        ModeSelect::getSupportedModesManager()->getModeOptionByMode(endpointId, newMode, &modeOptionPtr);
-    if (Status::Success != checkSupportedModeStatus)
+    using Traits = NumericAttributeTraits<uint8_t>;
+    if (!Traits::CanRepresentValue(/* isNullable = */ true, value))
     {
-        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "ModeSelect: Failed to find the option with mode %u", newMode);
-        commandHandler->AddStatus(commandPath, checkSupportedModeStatus);
-        return true;
+        return EMBER_ZCL_STATUS_CONSTRAINT_ERROR;
     }
-    ModeSelect::Attributes::CurrentMode::Set(endpointId, newMode);
+    Traits::StorageType storageValue;
+    Traits::WorkingToStorage(value, storageValue);
+    uint8_t * writable = Traits::ToAttributeStoreRepresentation(storageValue);
+    return emberAfWriteAttribute(endpointId, clusterId, ModeSelect::Attributes::StartUpMode::Id, writable,
+                                 ZCL_INT8U_ATTRIBUTE_TYPE);
+}
 
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "ModeSelect: ChangeToMode successful");
-    commandHandler->AddStatus(commandPath, Status::Success);
-    return true;
+EmberAfStatus Instance::SetStartUpModeNull() const
+{
+    using Traits = NumericAttributeTraits<uint8_t>;
+    Traits::StorageType value;
+    Traits::SetNull(value);
+    uint8_t * writable = Traits::ToAttributeStoreRepresentation(value);
+    return emberAfWriteAttribute(endpointId, clusterId, ModeSelect::Attributes::StartUpMode::Id, writable,
+                                 ZCL_INT8U_ATTRIBUTE_TYPE);
 }
 
-/**
- * Callback for Mode Select Cluster Server Initialization.
- * Enabled in src/app/zap-templates/templates/app/helper.js
- * @param endpointId    id of the endpoint that is being initialized
- */
-void emberAfModeSelectClusterServerInitCallback(EndpointId endpointId)
+std::map<uint32_t, Instance*> Instance::ModeSelectAliasesInstanceMap;
+
+CHIP_ERROR Instance::Init()
 {
+    // Check that the cluster ID given is a valid mode select alias cluster ID.
+    if (!std::any_of(AliasedClusters.begin(), AliasedClusters.end(), [this](ClusterId i){return i == clusterId;}))
```

Comment:
Capturing this-closures should be avoided on embedded code.

Reply:
What do you think of [this](https://github.com/project-chip/connectedhomeip/pull/26447/commits/d07fe9b74ee71319d19a0dd20490e3e4f087d7ed) solution?